### DISCUSSION TITLE
Почему не работает def-ttl 0 как в андроид приложении byedpi через gui-интерфейс?

### DISCUSSION THREAD

**Discussion by SaintAnd** [dovecoteescapee/ByeDPIAndroid#187] [General]
Здравствуйте! Почему не работает def-ttl 0 как в андроид приложении byedpi через gui-интерфейс?
Эта настройка очень важная, так как только при --def-ttl 0 работает обход блокировки в gui режиме. А я хочу через командную строку запускать эту программу? Писал об этой проблеме здесь  https://github.com/hufrea/byedpi/issues/204

**Comment by SaintAnd** [comment]
![380459356-c46c984f-8925-40fc-9cb5-3a1cc145c352](https://github.com/user-attachments/assets/7c80c8bb-40d1-41a5-b666-b4af4c94b3c1)

**Comment by zimbabwe303** [comment]
Нашёл ответ в коде [тут](https://github.com/dovecoteescapee/ByeDPIAndroid/blob/43db0955de687d43f9473374dc7c5548eaa55359/app/src/main/cpp/native-lib.c#L120) и [тут](https://github.com/hufrea/byedpi/blob/0a06ba41c9a48b1fa4471ac938fbc7d7080bf291/main.c#L348). Нулевой "Default TTL" означает, что значение TTL будет взято из дефолтных параметров для конкретной ОС при помощи функции [getsockopt](https://man7.org/linux/man-pages/man2/getsockopt.2.html). В версии byedpi для командной строки структура с параметрами изначально очищается нулями, поэтому отсутствие параметра `-g` [производит то же действие](https://github.com/hufrea/byedpi/blob/0a06ba41c9a48b1fa4471ac938fbc7d7080bf291/main.c#L949). Если у ОС ядро Linux и в его настойках ничего не меняли, то ip_default_ttl равно 64 (см. [тут](https://man7.org/linux/man-pages/man7/ip.7.html)). Следовательно, "Default TTL" равное **0**, в большинстве случаев, означает **64** (или иное дефолтное значение для конкретной системы).
